controllers/cloud.redhat.com/providers/kafka/strimzi.go

```diff
@@ -296,18 +335,88 @@ func (s *strimziProvider) Provide(app *crd.ClowdApp, c *config.AppConfig) error
 		}
 	}
 
+	ku := &strimzi.KafkaUser{}
+	nn := types.NamespacedName{
+		Name:      getKafkaUsername(s.Env, app),
+		Namespace: getKafkaNamespace(s.Env),
+	}
+
+	s.Cache.Create(KafkaUser, nn, ku)
+	labeler := utils.GetCustomLabeler(
+		map[string]string{"strimzi.io/cluster": s.Env.Spec.Providers.Kafka.Cluster.Name}, nn, s.Env,
+	)
+
+	labeler(ku)
+
+	ku.Spec = &strimzi.KafkaUserSpec{
+		Authentication: &strimzi.KafkaUserSpecAuthentication{
+			Type: strimzi.KafkaUserSpecAuthenticationTypeScramSha512,
+		},
+		Authorization: &strimzi.KafkaUserSpecAuthorization{
+			Acls: []strimzi.KafkaUserSpecAuthorizationAclsElem{},
+			Type: strimzi.KafkaUserSpecAuthorizationTypeSimple,
+		},
+	}
+
 	// update s.Config.Topics
-	if err := s.processTopics(app); err != nil {
+	if err := s.processTopics(app, ku); err != nil {
 		return err
 	}
 
+	s.Cache.Update(KafkaUser, ku)
+
+	for _, broker := range s.Config.BrokersTLS {
```

Comment:
For readability, can we move this logic into another func named something like `setBrokerCredentials` (I think that is the spirit of what's being done here?)

Reply:
We can